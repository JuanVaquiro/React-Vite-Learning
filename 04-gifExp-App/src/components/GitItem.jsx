import PropTypes  from "prop-types"

const GitItem = ({ title, url, id }) => {
   
    // console.log({title, url, id})
    return (
    <div className="item-card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

GitItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GitItem