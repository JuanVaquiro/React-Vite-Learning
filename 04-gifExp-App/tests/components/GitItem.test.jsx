import { render, screen } from "@testing-library/react";
import GitItem from "../../src/components/GitItem";

describe('test en el compoenntes <GitItem />', () => {

    const title = "Gifs";
    const url = 'https://gifs.com/gif.png'

    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render(<GitItem  title={ title } url={ url } />)
        
        expect( container ).toMatchSnapshot()
    
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
        render(<GitItem  title={ title } url={ url } />);
        // expect( screen.getByRole('img').src ).toBe( url )

        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('debe de mostrar el titulo en el componente', () => {
        
        render(<GitItem  title={ title } url={ url } />);
        expect( screen.getByText( title ) ).toBeTruthy();
    });

});