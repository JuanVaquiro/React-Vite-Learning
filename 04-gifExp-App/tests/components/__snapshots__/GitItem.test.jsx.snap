// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en el compoenntes <GitItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="item-card"
  >
    <img
      alt="Gifs"
      src="https://gifs.com/gif.png"
    />
    <p>
      Gifs
    </p>
  </div>
</div>
`;
